decibel-cpp/include/decibel/messaging/ProtobufProtocolFactory.h

```diff
@@ -51,7 +51,7 @@ class ProtobufProtocolFactory : public IProtocolFactory
         {
             auto protocol = std::make_unique<ProtocolT>(mDispatcher);
             protocol->RegisterSendErrorCallback(*mSendErrorCallback);
-            return protocol;
+            return std::move(protocol);
```

Comment:
The macOS headers refuse to move construct the `unique_ptr` without the `std::move` here, and barfs on the deleted copy constructor of `unique_ptr`'s. Given this is a pointer anyway, I don't think the lack of RVO would be too harmful